File: teste-pratico-java/src/main/java/com/heitorsilva/testepratico/model/Funcionario.java
package com.heitorsilva.testepratico.model;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonProperty.Access;

@Entity
public class Funcionario {
	
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Id
	private Long id;
	
	@Column(nullable = false, length = 40)
	private String nome;
	
	@Column(name = "cpf", nullable = false, length = 14, unique = true)
	private String cpf;

	@JsonFormat(pattern = "dd/MM/yyyy")
	@Temporal(TemporalType.TIMESTAMP)
	private Date dataNascimento;

	@Column(unique = true, length = 12)
	private String login;
	
	@Column(length = 12)
	@JsonProperty(access = Access.WRITE_ONLY)
	private String senha;

	public Long getId() {
		return id;
	}

	public String getNome() {
		return nome;
	}

	public String getCpf() {
		return cpf;
	}

	public Date getDataNascimento() {
		return dataNascimento;
	}

	public String getLogin() {
		return login;
	}

	public String getSenha() {
		return senha;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public void setCpf(String cpf) {
		this.cpf = cpf;
	}

	public void setDataNascimento(Date dataNascimento) {
		this.dataNascimento = dataNascimento;
	}

	public void setLogin(String login) {
		this.login = login;
	}

	public void setSenha(String senha) {
		this.senha = senha;
	}
	
}

File: teste-pratico-angular/src/app/shared/models/funcionario.ts
export interface IFuncionario {
  nome: string
  cpf: string
  dataNascimento: string
  login: string
  senha: string
}

export class Funcionario implements IFuncionario {
  nome: string
  cpf: string
  dataNascimento: string
  login: string
  senha: string

  constructor(funcionario : IFuncionario) {
    this.nome = funcionario?.nome ? funcionario.nome : ''
    this.cpf = funcionario?.cpf ? funcionario.cpf : ''
    this.dataNascimento = funcionario?.dataNascimento ? funcionario.dataNascimento : ''
    this.login = funcionario?.login ? funcionario.login : ''
    this.senha = funcionario?.senha ? funcionario.senha : ''
  }
}
